service/src/main/java/edu/tamu/catalog/service/FolioCatalogService.java

```diff
@@ -276,4 +285,63 @@ private void buildCoreItem(String instanceId, String barcode, NodeList nodes, Ma
         catalogItems.put(barcode, itemData);
     }
 
+    @Override
+    public FeesFines getFeesFines(String uin) throws ParseException {
+        String path = "patron/account/";
+        String additional = "&includeLoans=false&includeCharges=true&includeHolds=false";
+        String url = String.format("%s%s%s?apikey={apikey}%s", getAPIBase(), path, uin, additional);
+        String apiKey = getAuthentication().get(CatalogServiceFactory.FIELD_APIKEY);
+
+        logger.debug(String.format("Asking for fines from: %s", url));
+
+        JsonNode node = restTemplate.getForObject(url, JsonNode.class, apiKey);
```

Comment:
If DEBUG logging is turned on for `org.springframework` this will log the entire URL which almost defeats the purpose of attempting to mask the apiKey with URL log above.

Reply:
Substitution of `{apikey}` doesn't happen until `restTemplate.getForObject(url, JsonNode.class, apiKey);` and then it happens within the `getForObject` and not on the `url` string variable at all.

Reply:
Yes. `getForObject` call is what templates the URL. What I was just noting is that if certain debug logs are enabled, it will log the entire URL with apiKey and all.

Reply:
Honestly, the logs are as secure as the application.properties file and the apiKey is in the application.properties. If we eventually pulled logs into Splunk or other CLM, it would be best for it not be in the logs.